pkgs/applications/misc/retool/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, python3
+, fetchFromGitHub
+, qt6
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "retool";
+  version = "2.01.5";
+
+  format = "pyproject";
+  disabled = python3.pkgs.pythonOlder "3.10";
+
+  src = fetchFromGitHub {
+    owner = "unexpectedpanda";
+    repo = "retool";
+    rev = "v${version}";
+    hash = "sha256-L1jcsV4qGj5uhQwEw2Z4w3wNDWNbbABsPFTwdfQ0r54=";
+  };
+
+  # TODO: remove once this PR is merged:
+  # https://github.com/unexpectedpanda/retool/pull/286
+  patches = [
+    ./encapsulate_code_in_main_retoolgui.patch
+    ./fix_pyproject.patch
+    # pythonRelaxDepsHook is not working for some reason
+    ./relax_deps.patch
```

Comment:
Found the issue, for some reason the wheel ends up with `2.1.5` name instead of `2.01.5`. Not sure why it happens, seems to be a bug in Poetry (because the version upstream is declared correctly in the `pyproject.toml` file).

This will be fixed once https://github.com/NixOS/nixpkgs/pull/248516 hits `master`, but right now it is in `staging-next`. If `staging-next` gets merged before this PR is reviewed, I can just rebase and remove this patch.